File: TextosJapones/src/main/java/org/jisho/textosJapones/components/NumberToStringConverter.java
package org.jisho.textosJapones.components;

import java.math.BigDecimal;
import java.text.NumberFormat;
import java.util.Locale;

import javafx.util.converter.NumberStringConverter;

public class NumberToStringConverter extends NumberStringConverter {
    public NumberToStringConverter() {
        super();
    }

    public NumberToStringConverter(Locale locale, String pattern) {
        super(locale, pattern);
    }

    public NumberToStringConverter(Locale locale) {
        super(locale);
    }

    public NumberToStringConverter(NumberFormat numberFormat) {
        super(numberFormat);
    }

    public NumberToStringConverter(String pattern) {
        super(pattern);
    }

    @Override
    public Number fromString(String value) {
        //to transform the double, given by the textfield, just multiply by 100 and round if any left
        Number rValue = Math.round(super.fromString(value).doubleValue() * 100);
        return rValue.longValue();
    }

    @Override
    public String toString(Number value) {
        if(value == null) {
            return "";
        }
        //Check for too big long value
        //If the long is too big, it could result in a strange double value.
        if(value.longValue() > 1000000000000l || value.longValue() < -1000000000000l ) {
            return "";
        }
        BigDecimal myBigDecimal = new BigDecimal(value.longValue());
        //to convert the long to a double (currency with fractional digits)
        myBigDecimal = myBigDecimal.movePointLeft(2);
        double asDouble = myBigDecimal.doubleValue();
        if(asDouble == Double.NEGATIVE_INFINITY || asDouble == Double.POSITIVE_INFINITY) {
            return "";
        }
        return super.toString(asDouble);
    }
}

File: ProcessaTexto/src/main/kotlin/br/com/fenix/processatexto/components/NumberToStringConverter.kt
package br.com.fenix.processatexto.components

import javafx.util.converter.NumberStringConverter
import java.math.BigDecimal
import java.text.NumberFormat
import java.util.*

class NumberToStringConverter : NumberStringConverter {

    constructor() : super() {}
    constructor(locale: Locale, pattern: String) : super(locale, pattern) {}
    constructor(locale: Locale) : super(locale) {}
    constructor(numberFormat: NumberFormat) : super(numberFormat) {}
    constructor(pattern: String) : super(pattern) {}

    @Override
    override fun fromString(value: String): Number {
        //to transform the double, given by the textfield, just multiply by 100 and round if any left
        val rValue: Number = Math.round(super.fromString(value).toDouble() * 100)
        return rValue.toLong()
    }

    @Override
    override fun toString(value: Number): String {
        if (value == null) {
            return ""
        }
        //Check for too big long value
        //If the long is too big, it could result in a strange double value.
        if (value.toLong() > 1000000000000L || value.toLong() < -1000000000000L) {
            return ""
        }
        var myBigDecimal = BigDecimal(value.toLong())
        //to convert the long to a double (currency with fractional digits)
        myBigDecimal = myBigDecimal.movePointLeft(2)
        val asDouble: Double = myBigDecimal.toDouble()
        return if (asDouble == Double.NEGATIVE_INFINITY || asDouble == Double.POSITIVE_INFINITY) {
            ""
        } else
            super.toString(asDouble)
    }
}